andela/banshee-bn-backend
# #167891570: Setup Docker for backend api

#### What does this PR do?

- Setup docker for local development

#### Description of Task to be completed?

- build docker image for node

- build/pull docker image for postgresql

- compose `docker-compose.yml` for managing the containers

- setup the environment variable for the images

#### How should this be manually tested?

- ensure that [docker](https://www.docker.com/) is installed your local machine

- clone the repo to a directory and `cd` into the directory

- bootup your `docker-machine` using `docker-machine start machineName` and obtain the `IP address` of the machine

- rename `.env.example` to `.env` substituting each variable with its value

- setup the terminal to connect to `docker daemon`

- in the root of the app, run `docker-compose build` to build the node image

- if above is successful, run `docker-compose up` to bring all the containers

- on postman or browser, visit the `docker-machine ip:app-port/` to make request to the app

#### Any background context you want to provide?

- docker setup was built to eliminate the challenge of differences in development platforms

#### What are the relevant pivotal tracker stories?

- [#167891570](https://www.pivotaltracker.com/story/show/167891570)

#### Screenshots (if appropriate)

- N/A

#### Questions:

- how to run commands inside the database container?
__run `docker exec -it containerName /bin/bash`__
